src/machines/sketchSolve/sketchSolveMode.ts

```diff
@@ -55,38 +56,57 @@ export const sketchSolveMachine = setup({
     'send updated selection to move tool': sendTo('moveTool', {
       type: 'update selection',
     }),
-    'spawn tool': assign(({ event, spawn }) => {
+    'store pending tool': assign(({ event }) => {
       assertEvent(event, 'equip tool')
+      return { pendingTool: event.data.tool }
+    }),
+    'spawn tool': assign(({ event, spawn, context }) => {
+      // Determine which tool to spawn based on event type
+      let toolToSpawn: EquipTool
+
+      if (event.type === 'equip tool') {
+        toolToSpawn = event.data.tool
+      } else if (
+        event.type === 'xstate.done.actor.tool' &&
+        context.pendingTool
+      ) {
+        toolToSpawn = context.pendingTool
+      } else {
+        console.error('Cannot determine tool to spawn')
+        return {}
+      }
+
       let toolActor
-      switch (event.data.tool) {
-        case 'dimension':
-          toolActor = spawn('dimensionToolActor', { id: 'tool' })
+      switch (toolToSpawn) {
+        case 'dimensionTool':
+          toolActor = spawn(toolToSpawn, { id: 'tool' })
+          break
+        case 'centerRectTool':
+          toolActor = spawn(toolToSpawn, { id: 'tool' })
           break
-        case 'center rectangle':
-          toolActor = spawn('centerRectToolActor', { id: 'tool' })
+        case 'pointTool':
+          toolActor = spawn(toolToSpawn, { id: 'tool' })
           break
         default:
-          const _exhaustiveCheck: never = event.data.tool
+          const _exhaustiveCheck: never = toolToSpawn
       }
+      console.log('spawned tool?')
 
       return {
         toolActor,
-        sketchSolveTool: event.data.tool,
+        sketchSolveTool: toolToSpawn,
```

Comment:
Future note, might be worth naming this `sketchToolName`, as differentiated from `*Actor`, and assuming we drop `Solve` once this is the standard sketch mode.